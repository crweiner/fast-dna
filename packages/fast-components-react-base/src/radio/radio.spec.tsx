import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import Radio, {
    IRadioClassNameContract,
    IRadioHandledProps,
    IRadioManagedClasses,
    IRadioUnhandledProps,
    RadioProps,
    RadioSlot
} from "./radio";
import Label from "../label";

/**
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("radio snapshots", (): void => {
    generateSnapshots(examples);
});

describe("radio", (): void => {
    const Component: React.ComponentClass<IRadioHandledProps & IRadioManagedClasses> = examples.component;
    const managedClasses: IRadioClassNameContract = {
        radio_input: "radio_input"
    };

    const inputSelector: string = `.${managedClasses.radio_input}`;

    test("should have a displayName that matches the component name", () => {
        expect((Radio as any).name).toBe(Radio.displayName);
    });

    test("should call a registerd callback after a change event", () => {
        const onChange: any = jest.fn();
        const controlled: any = shallow(
            <Component managedClasses={managedClasses} checked={true} onChange={onChange}>
                <div slot={RadioSlot.label} />
            </Component>
        );
        const uncontrolled: any = shallow(
            <Component managedClasses={managedClasses} onChange={onChange} />
        );

        controlled.find(inputSelector).simulate("change");

        expect(onChange).toHaveBeenCalledTimes(1);

        uncontrolled.find(inputSelector).simulate("change");

        expect(onChange).toHaveBeenCalledTimes(2);
    });
});
