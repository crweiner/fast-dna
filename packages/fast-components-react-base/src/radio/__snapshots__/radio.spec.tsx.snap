// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio snapshots Radio: 0 1`] = `
<div
  className="radio"
>
  <input
    checked={true}
    className="radio_input"
    defaultChecked={false}
    disabled={null}
    id={undefined}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="radio_stateIndicator"
  />
</div>
`;

exports[`radio snapshots Radio: 1 1`] = `
<div
  className="radio radio__disabled"
>
  <input
    checked={null}
    className="radio_input"
    defaultChecked={false}
    disabled={true}
    id={undefined}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="radio_stateIndicator"
  />
</div>
`;
